bellman/src/gadgets/boolean.rs

```diff
@@ -412,24 +412,24 @@ impl Boolean {
     }
 
     pub fn get_value(&self) -> Option<bool> {
-        match self {
-            &Boolean::Constant(c) => Some(c),
-            &Boolean::Is(ref v) => v.get_value(),
-            &Boolean::Not(ref v) => v.get_value().map(|b| !b),
+        match *self {
```

Comment:
The clippy warning that this change addressed is:
```
warning: you don't need to add `&` to all patterns
   --> bellman/src/gadgets/boolean.rs:415:9
    |
415 | /         match self {
416 | |             &Boolean::Constant(c) => Some(c),
417 | |             &Boolean::Is(ref v) => v.get_value(),
418 | |             &Boolean::Not(ref v) => v.get_value().map(|b| !b),
419 | |         }
    | |_________^
    |
    = note: #[warn(clippy::match_ref_pats)] on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_ref_pats
help: instead of prefixing all patterns with `&`, you can dereference the expression
    |
415 |         match *self {
416 |             Boolean::Constant(c) => Some(c),
417 |             Boolean::Is(ref v) => v.get_value(),
418 |             Boolean::Not(ref v) => v.get_value().map(|b| !b),
    |
```
Dereferencing is still necessary here:
```
$ git diff
diff --git a/bellman/src/gadgets/boolean.rs b/bellman/src/gadgets/boolean.rs
index b26bb19..dafa82f 100644
--- a/bellman/src/gadgets/boolean.rs
+++ b/bellman/src/gadgets/boolean.rs
@@ -412,7 +412,7 @@ impl Boolean {
     }

     pub fn get_value(&self) -> Option<bool> {
-        match *self {
+        match self {
             Boolean::Constant(c) => Some(c),
             Boolean::Is(ref v) => v.get_value(),
             Boolean::Not(ref v) => v.get_value().map(|b| !b),
$ cargo build
   Compiling bellman v0.1.0 (/.../librustzcash/bellman)
error[E0308]: mismatched types
   --> bellman/src/gadgets/boolean.rs:416:42
    |
416 |             Boolean::Constant(c) => Some(c),
    |                                          ^
    |                                          |
    |                                          expected bool, found &bool
    |                                          help: consider dereferencing the borrow: `*c`
    |
    = note: expected type `bool`
               found type `&bool`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: Could not compile `bellman`.

To learn more, run the command again with --verbose.
$ rustc --version
rustc 1.36.0 (a53f9df32 2019-07-03)
```